lib/ts/logger.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import debug from "debug";
+import { version } from "./version";
+
+/*
+ The debug logger and info logger defined below use the debug lib to log debug and info messages when the DEBUG env is set with the com.supertokens namespace.
+For example:
+    Adding a info log to the emailpassword signin api
+    Adding a debug log to the SignIn API which will print the status when a non OK response is returned
+Flow: 
+    SignIn with with valid credentials;
+    SignIn with invalid credentials
+Output: (with DEBUG=com.supertokens:*)
+  com.supertokens:info {t: "1647340361090", msg: "SignInAPI is called", code: 2, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:54:69", sdkVer: "9.1.0"} +0ms
+  com.supertokens:debug {t: "1647340361295", msg: "SignInAPI replied with status: OK", code: 1, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:65:76", sdkVer: "9.1.0"} +0ms
+  com.supertokens:info {t: "1647340563421", msg: "SignInAPI is called", code: 2, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:54:69", sdkVer: "9.1.0"} +0ms
+  com.supertokens:debug {t: "1647340563565", msg: "SignInAPI replied with status: WRONG_CREDENTIALS_ERROR", code: 1, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:71:76", sdkVer: "9.1.0"} +0ms
+*/
+
+export let loggerCodes = {
```

Comment:
> What are these codes used for?

The initial thought behind codes was that if there were common issues between users and we found common patterns in the debug logs we could have a section in our docs for troubleshooting like "Hey if you got this `sequence of codes` this is how you fix it". 

> It should be a const or even better, an enum. In general, you should prefer const to let.

Yup will make the change

Reply:
A "sequence of code" sounds a bit hard to search. I thought they were for looking up more info about each log item (that's how I've seen these used, or localized messages). That may come in handy if/when we have some harder-to-understand logs where users need more explanation.
Anyway, it doesn't complicate things too much, and it may be helpful; I'm just curious. I'll resolve this after the second point is addressed in the code.